tests/phpunit/integration/Modules/Reader_Revenue_ManagerTest.php

```diff
@@ -73,4 +111,57 @@ public function test_service_classes_exist() {
 			class_exists( 'Google\Site_Kit_Dependencies\Google\Service\SubscribewithGoogle' )
 		);
 	}
+
+	public function test_get_data_endpoints() {
+		$this->assertEqualSets(
+			array(
+				'publications',
+			),
+			$this->reader_revenue_manager->get_datapoints()
+		);
+	}
+
+	public function test_service_entity_has_access() {
+		FakeHttp::fake_google_http_handler(
+			$this->reader_revenue_manager->get_client(),
+			function ( Request $request ) {
+				$url = parse_url( $request->getUri() );
+
+				switch ( $url['path'] ) {
+					case '/v1/publications':
+						return new Response( 200 );
+				}
+			}
+		);
+
+		$this->reader_revenue_manager->register();
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->authentication->get_oauth_client()->get_required_scopes()
+		);
+
+		$data = $this->reader_revenue_manager->get_data( 'publications' );
+		$this->assertNotWPError( $data );
+	}
+
+	public function test_service_entity_has_no_access() {
+		FakeHttp::fake_google_http_handler(
+			$this->reader_revenue_manager->get_client(),
+			function ( Request $request ) {
+				$url = parse_url( $request->getUri() );
+
+				switch ( $url['path'] ) {
+					case '/v1/publications':
+						return new Response( 200 );
+				}
+			}
+		);
+
+		$this->reader_revenue_manager->register();
+		$data = $this->reader_revenue_manager->get_data( 'publications' );
+		$this->assertWPError( $data );
+
+		$error_data = $data->get_error_data();
+		$this->assertEquals( 403, $error_data['status'] );
+	}
```

Comment:
We have built-in test coverage for service entity access, using `Module_With_Service_Entity_ContractTests` trait. Please see its usage in `tests/phpunit/integration/Modules/Analytics_4Test.php` for example.